lib/DxilContainer/DxcContainerBuilder.cpp

```diff
@@ -164,11 +166,65 @@ DxcContainerBuilder::SerializeContainer(IDxcOperationResult **ppResult) {
         {DxcOutputObject::DataOutput(DXC_OUT_OBJECT, pResult, DxcOutNoName),
          DxcOutputObject::DataOutput(DXC_OUT_ERRORS, pErrorBlob, DxcOutNoName)},
         ppResult));
+
+    // Add Hash.
+    if (ppResult != nullptr && *ppResult != nullptr) {
+      HRESULT HR;
+      (*ppResult)->GetStatus(&HR);
+      if (SUCCEEDED(HR)) {
+        CComPtr<IDxcBlob> pObject;
+        HR = (*ppResult)->GetResult(&pObject);
+        if (SUCCEEDED(HR)) {
+          LPVOID PTR = pObject->GetBufferPointer();
+          if (IsDxilContainerLike(PTR, pObject->GetBufferSize()))
+            HashAndUpdate((DxilContainerHeader *)PTR);
+        }
+      }
+    }
```

Comment:
We should deep nesting.  The rest of this function uses IFT rather than nested if (SUCCEEDED)'s.

```suggestion
    if (ppResult != nullptr && *ppResult != nullptr) {
      HRESULT HR;
      (*ppResult)->GetStatus(&HR);
      IFT(HR);

      CComPtr<IDxcBlob> pObject;
      IFT((*ppResult)->GetResult(&pObject));
      LPVOID PTR = pObject->GetBufferPointer();
      if (IsDxilContainerLike(PTR, pObject->GetBufferSize()))
        HashAndUpdate((DxilContainerHeader *)PTR);
    }
```

Reply:
Updated.

Reply:
IFT will throw exception. The original code doesn't throw.

Reply:
Yeah, the pattern used throughout this function is that HRESULTs that we want to return as errors get thrown and then caught by CATCH_CPP_RETURN_HRESULT.  It's odd to have everything in SerializeContainer use the IFT approach for this and then suddenly switch to a different pattern 60 lines through it.

Although now I've read your latest version of this it is clear to me that the intent is to ignore any failures while trying to write out the hash?  Why's that?  Surely SerializeContainer has failed to do what it was asked to do if it couldn't add the hash.

Reply:
If there's a good reason for ignoring the errors it'd be good to have that explained in a comment here.

Reply:
After remove IFT when verify root signature in dxc (that will allow dxc to check the error), return failed hr works.

Reply:
I'm not sure we understand each other here, but it's at least a bit closer to the existing style in that function now.